### PR TITLE
Update hls.js types

### PR DESCRIPTION
**Pull request by hamidfzm**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/video-dev/hls.js/blob/master/docs/API.md#subtitle-tracks-control-api & https://github.com/video-dev/hls.js/blob/master/docs/API.md#audio-tracks-control-api
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## hls.js ([<kbd>unpkg</kbd>](https://unpkg.com/browse/hls.js@latest/))
was missing the following properties:
1. prefixed
1. EventEmitter
1. Hls
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against cfddd9d18742d8d60965658b5e3066f0f73b1de1
</p>

File: types/hls.js/index.d.ts [hunk 1]
```diff
@@ -354,12 +354,28 @@ declare namespace Hls {
     }
 
     interface AudioTrack {
```
Comment (Beraliv) [inline types/hls.js/index.d.ts:356]: AudioTrack as mentioned in previous PR, was removed from TS and you don't need to leave it as is
Probably, the best solution here is to add `HlsAudioTrack`

Related PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43335

File: types/hls.js/index.d.ts [hunk 2]
```diff
@@ -1714,10 +1730,20 @@ declare class Hls {
      * position of live sync point (ie edge of live position minus safety delay defined by hls.config.liveSyncDuration)
      */
     readonly liveSyncPosition: number;
+    /**
+     * get : array of audio tracks exposed in manifest
+     */
+    readonly audioTracks: Hls.AudioTrack[];
```
Comment (Beraliv) [inline types/hls.js/index.d.ts:1736]: Could you please add `HlsAudioTrack` and use it here?

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43335#issuecomment-603298347

File: types/hls.js/index.d.ts (hunk 1 above)
Comment (hamidfzm) [reply types/hls.js/index.d.ts]: Applied requested changes. Is it ok now?

**Comment by hamidfzm** [comment]
@Beraliv Could you please tell me why tests are failing? I didn't apply major changes to break tests.

File: types/hls.js/index.d.ts (hunk 1 above)
Comment (Beraliv) [reply types/hls.js/index.d.ts]: LGTM

**Comment by Beraliv** [comment]
> @Beraliv Could you please tell me why tests are failing? I didn't apply major changes to break tests.

I found theses errors:

```
Error in hls.js
Error: /home/runner/work/DefinitelyTyped/DefinitelyTyped/types/hls.js/index.d.ts:1716:1
ERROR: 1716:1  jsdoc-format  jsdoc is not formatted correctly on this line
ERROR: 1716:1  jsdoc-format  asterisks in jsdoc must be aligned
ERROR: 1717:1  jsdoc-format  jsdoc is not formatted correctly on this line
```

**Review by Beraliv — APPROVED** [review]
👍

**Review by drax98 — APPROVED** [review]

**Review by jgainfort — APPROVED** [review]

File: types/hls.js/index.d.ts [hunk 3]
```diff
@@ -353,13 +353,29 @@ declare namespace Hls {
         error?: CustomLogger;
     }
 
-    interface AudioTrack {
```
Comment (adripanico) [inline types/hls.js/index.d.ts:356]: Why are you changing the interface name? This is a breaking change for anybody explicitly using this interface.

**Review by adripanico — CHANGES_REQUESTED** [review]

File: types/hls.js/index.d.ts (hunk 3 above)
Comment (jgainfort) [reply types/hls.js/index.d.ts:356]: See https://github.com/DefinitelyTyped/DefinitelyTyped/pull/43335

**Comment by adripanico** [comment]
> Please fill in this template.
> 
>     * [x]  Use a meaningful title for the pull request. Include the name of the package modified.
> 
>     * [x]  Test the change in your own code. (Compile and run.)
> 
>     * [ ]  Add or edit tests to reflect the change. (Run with `npm test`.)
> 
>     * [x]  Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
> 
>     * [x]  Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
> 
>     * [x]  Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
> 
> 
> Select one of these and delete the others:
> 
> If adding a new definition:
> 
>     * [ ]  The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
> 
>     * [ ]  If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
> 
>     * [ ]  Create it with `dts-gen --dt`, not by basing it on an existing project.
> 
>     * [ ]  Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
> 
>     * [ ]  `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
> 
>     * [ ]  `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.
> 
> 
> If changing an existing definition:
> 
>     * [ ]  Provide a URL to documentation or source code which provides context for the suggested changes: <>
> 
>     * [ ]  If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
> 
>     * [ ]  Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
> 
>     * [ ]  If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.
> 
> 
> If removing a declaration:
> 
>     * [ ]  If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
> 
>     * [ ]  Delete the package's directory.
> 
>     * [ ]  Add it to `notNeededPackages.json`.

Thanks for your PR.

Please, fulfill the PR form and remove those elements when applicable. It's is difficult to review the changes is you no provide the needed information.

**Comment by hamidfzm** [comment]
@adripanico What should I add to this PR form or code? Can you help me since it's my first time creating a pull request for this repo?

File: types/hls.js/index.d.ts (hunk 3 above)
Comment (adripanico) [reply types/hls.js/index.d.ts:356]: Ok. Looks good to me then.

**Comment by adripanico** [comment]
> @adripanico What should I add/modify to this PR form or code? Can you help me since it's my first time creating a pull request for this repo?

Sure, I understand. 

When it says "Select one of these and delete the others:" you should keep one of the below sections and removed the rest. In your case you should keep the section "If changing an existing definition:".

Then, you should also fill the items of this list (for example, for the first point, add proper URLs with the documentation or issues that lead to this PR).

**Review by adripanico — APPROVED** [review]

**Comment by hamidfzm** [comment]
@adripanico Done.

**Comment by adripanico** [comment]
Ready to merge

**Pull request merged**